community/kernel/src/main/java/org/neo4j/kernel/configuration/Config.java

```diff
@@ -153,28 +271,83 @@ public Config with( Map<String, String> additionalConfig, Class<?>... settingsCl
      *
      * @param changes settings to add and override
      */
-    public Config augment( Map<String, String> changes )
+    public Config augment( Map<String,String> changes )
     {
-        Map<String, String> params = getParams();
+        Map<String,String> params = new HashMap<>( this.params );
         params.putAll( changes );
         replaceSettings( params );
         return this;
     }
 
-    public Iterable<Class<?>> getSettingsClasses()
-    {
-        return settingsClasses;
-    }
-
+    /**
+     * Replays possible existing messages into this log
+     *
+     * @param log to use
+     */
```

Comment:
The main function of this method is assigning the `log` field. The replying of log messages is a side-effect, if the previous value was a `BufferingLog`.